### PR TITLE
AEA-928 Allow finer-grained overwrite of component configuration from aea-config.yaml.

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Currently, it is not possible to overwrite configurations _only_ for specific fields from the `aea-config.yaml` file, as could be done with `aea set`.

This PR allows that. (TODO: add an example)

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: aea/configurations/loader.py [hunk 1]
```diff
@@ -77,37 +76,74 @@ def make_jsonschema_base_uri(base_uri_path: Path) -> str:
...
-    def __init__(self, schema_filename: str, configuration_class: Type[T]):
+    :param component_type: a component type.
+    :return: the path to the JSON schema file.
+    """
+    path_prefix: Path = Path(_SCHEMAS_DIR) / "configurable_parts"
+    if component_type in {ComponentType.SKILL, ComponentType.CONNECTION}:
+        filename_prefix = component_type.value
+    else:
+        filename_prefix = "base"
+    full_path = path_prefix / (filename_prefix + "-custom_config.json")
```
Comment (DavidMinarsch) [inline aea/configurations/loader.py:91]: `"-custom_config.json"` should be declared as constant top of file

File: aea/configurations/loader.py [hunk 2]
```diff
@@ -77,37 +76,74 @@ def make_jsonschema_base_uri(base_uri_path: Path) -> str:
...
-    """This class implement parsing, serialization and validation functionalities for the 'aea' configuration files."""
+def _get_path_to_custom_config_schema_from_type(component_type: ComponentType) -> str:
+    """
+    Get the path to the custom config schema
 
-    def __init__(self, schema_filename: str, configuration_class: Type[T]):
+    :param component_type: a component type.
+    :return: the path to the JSON schema file.
+    """
+    path_prefix: Path = Path(_SCHEMAS_DIR) / "configurable_parts"
```
Comment (DavidMinarsch) [inline aea/configurations/loader.py:86]: `"configurable_parts"` should be declared constant top of file

File: aea/configurations/loader.py [hunk 3]
```diff
@@ -77,37 +76,74 @@ def make_jsonschema_base_uri(base_uri_path: Path) -> str:
...
+    if component_type in {ComponentType.SKILL, ComponentType.CONNECTION}:
+        filename_prefix = component_type.value
+    else:
+        filename_prefix = "base"
```
Comment (DavidMinarsch) [inline aea/configurations/loader.py:90]: declare constant where appropriate and use here

File: aea/configurations/loader.py [hunk 4]
```diff
@@ -245,20 +281,33 @@ def _load_agent_config(self, file_pointer: TextIO) -> AgentConfig:
...
+    ) -> Dict[ComponentId, Dict]:
+        """
+        Get the component configurations from the tail pages of the aea-config.yaml file.
+
+        :param configuration_file_jsons: the JSON objects of the custom configurations of a aea-config.yaml file.
+        :return: a dictionary whose keys are component ids and values are the configurations.
+        """
         component_configurations: Dict[ComponentId, Dict] = {}
         # load the other components.
         for i, component_configuration_json in enumerate(configuration_file_jsons[1:]):
```
Comment (DavidMinarsch) [inline aea/configurations/loader.py:301]: 1 should be declared as constant, like `STARTING_INDEX_CUSTOM_CONFIGS` or similar

File: aea/configurations/loader.py [hunk 5]
```diff
@@ -278,7 +327,7 @@ def _dump_component_config(self, configuration: T, file_pointer: TextIO) -> None
 
     def _process_component_section(
         self, i: int, component_configuration_json: Dict
```
Comment (DavidMinarsch) [inline aea/configurations/loader.py:329]: We discourage one letter variables!

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by solarw — APPROVED** [review]
lgtm

**Pull request merged**